packages/typespec-azure-resource-manager/lib/decorators.tsp

```diff
@@ -191,3 +191,19 @@ extern dec resourceBaseType(
   target: Model,
   baseType: "Tenant" | "Subscription" | "ResourceGroup" | "Location" | "Extension"
 );
+
+/**
+ * This decorator is used to indicate the identifying properties of objects in the array, e.g. size
+ * The properties that are used as identifiers for the object needs to be provided as a list of strings.
+ *
+ * @param properties The list of properties that are used as identifiers for the object. This needs to be provided as a list of strings.
+ *
+ * @example
+ * ```typespec
+ * model Pet {
+ *  @identifiers(["size"])
+ *  dog: Dog;
+ * }
+ * ```
+ */
+extern dec identifiers(entity: ModelProperty, properties: string[]);
```

Comment:
make properties be `valueof string[]`

alternatively could we not have this be using the actual model property reference?

Reply:
```
@identifiers(Foo.id, Foo.secondId)
```

Reply:
if I remember correctly there was a problem with that approach for more complex scenarios, for example 
`@identifiers(Foo.id, Foo.secondId.otherId)`  wasn't able to resolve the `otherId` and this is supported for x-ms-identifiers, but I can check again this

Reply:
```
Foo.secondId::type.otherId
```
would technically be the right syntax but yeah wwe don't have a way to get the trace up.

But would change to `valueof` at least, it will simplify the js code and will not allow you to pass `string[]`